modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java

```diff
@@ -221,9 +226,44 @@ private String _buildFragmentHTML(
 		return sb.toString();
 	}
 
-	private String _getAPIURL(ObjectEntry fdsEntryObjectEntry) {
-		Map<String, Object> properties = fdsEntryObjectEntry.getProperties();
+	private String _embedURLParameters(String apiUrl, HttpServletRequest httpServletRequest){
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		Pattern pattern = Pattern.compile("\\{(.*?)\\}", Pattern.CASE_INSENSITIVE);
```

Comment:
The _pattern_ for cases where we need a regex (no pun intended) is to use a static final class attribute so that pattern gets compiled once at JVM initialization time, not each time we run the method. Something like:

```
	private static final Pattern _pattern = Pattern.compile("\\{(.*?)\\}", Pattern.CASE_INSENSITIVE);

```

then, one can directly use `_pattern.matcher()` from anywhere in the class.

We don't need a regex to find occurrences, see below

Reply:
Good to know!